src/main/java/com/google/api/codegen/transformer/InitCodeTransformer.java

```diff
@@ -51,106 +51,34 @@
-  }
-
-  public InitCodeView generateMockResponseObjectInitCode(
-      MethodTransformerContext context, SymbolTable table, TestValueGenerator valueGenerator) {
-    ArrayList<Field> primitiveFields = new ArrayList<>();
-    for (Field field : context.getMethod().getOutputMessage().getFields()) {
-      if (field.getType().isPrimitive() && !field.getType().isRepeated()) {
-        primitiveFields.add(field);
-      }
+  /** Generates initialization code from the given InitCodeTransformerContext object. */
+  public static InitCodeView generateInitCode(InitCodeTransformerContext context) {
+    InitCodeNode rootNode = createRootNode(context);
+    MethodTransformerContext methodContext = context.methodContext();
+    if (context.isFlattened()) {
+      return buildInitCodeViewFlattened(methodContext, rootNode);
+    } else {
+      return buildInitCodeViewRequestObject(methodContext, rootNode);
     }
-    InitCodeNode rootNode =
-        InitCodeNode.createTree(
-            InitTreeParserContext.newBuilder()
-                .table(table)
-                .valueGenerator(valueGenerator)
-                .rootObjectType(context.getMethod().getOutputType())
-                .initValueConfigMap(createCollectionMap(context))
-                .additionalSubTrees(createMockResponseAdditionalSubTrees(context))
-                .initFields(primitiveFields)
-                .suggestedName(Name.from("expected_response"))
-                .build());
-    return buildInitCodeViewRequestObject(context, rootNode);
   }
 
-  public InitCodeView generateSmokeTestInitCode(
-      MethodTransformerContext context, SymbolTable table) {
-    SmokeTestConfig testConfig = context.getInterfaceConfig().getSmokeTestConfig();
-    InitCodeNode rootNode =
-        InitCodeNode.createTree(
-            InitTreeParserContext.newBuilder()
-                .table(table)
-                .rootObjectType(testConfig.getMethod().getInputType())
-                .initValueConfigMap(createCollectionMap(context))
-                .initFieldConfigStrings(testConfig.getInitFieldConfigStrings())
-                .suggestedName(Name.from("request"))
-                .build());
-    return buildInitCodeViewFlattened(context, rootNode);
+  private static InitCodeNode createRootNode(InitCodeTransformerContext context) {
+    MethodTransformerContext methodContext = context.methodContext();
+    return InitCodeNode.createTree(
+        InitTreeParserContext.newBuilder()
+            .table(context.symbolTable())
+            .rootObjectType(context.initObjectType())
+            .initValueConfigMap(createCollectionMap(methodContext))
+            .initFieldConfigStrings(methodContext.getMethodConfig().getSampleCodeInitFields())
+            .initFields(context.fields())
+            .suggestedName(context.suggestedName())
+            .valueGenerator(context.valueGenerator())
+            .additionalSubTrees(context.additionalNodes())
+            .build());
```

Comment:
The InitCodeTransformerContext looks very similar to InitTreeParserContext - do we really need both?